compiling noiseprofile.c throws warnings

### Is there an existing issue for this?

- [x] I checked and did not find my issue in the already reported ones

### Describe the bug

Commit 83c2151 (Apr 27, 2023) by @kmilos tried to remove  4 warnings (`fscanf`, `fread`) but I still see them in current master. And two more ( `MIN`, `MAX` redefined) :

```
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:123: warning: "MIN" redefined
  123 | #define MIN(a,b) ((a>b)?b:a)
      | 
In file included from /usr/lib/x86_64-linux-gnu/glib-2.0/include/glibconfig.h:9,
                 from /usr/include/glib-2.0/glib/gtypes.h:34,
                 from /usr/include/glib-2.0/glib/galloca.h:34,
                 from /usr/include/glib-2.0/glib.h:32,
                 from /usr/include/glib-2.0/gmodule.h:30,
                 from /media/ph/ssd-120/darktable/master/tools/noise/../../src/common/dttypes.h:30,
                 from /media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:6:
/usr/include/glib-2.0/glib/gmacros.h:936: note: this is the location of the previous definition
  936 | #define MIN(a, b)  (((a) < (b)) ? (a) : (b))
      | 
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:124: warning: "MAX" redefined
  124 | #define MAX(a,b) ((a>b)?a:b)
      | 
/usr/include/glib-2.0/glib/gmacros.h:933: note: this is the location of the previous definition
  933 | #define MAX(a, b)  (((a) > (b)) ? (a) : (b))
      | 
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c: In function ‘read_pfm’:
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:62:10: warning: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   62 |   (void) fscanf(f, "PF\n%d %d\n%*[^\n]", wd, ht);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:66:10: warning: ignoring return value of ‘fread’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   66 |   (void) fread(p, sizeof(float)*3, (*wd)*(*ht), f);
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c: In function ‘read_histogram’:
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:81:12: warning: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   81 |     (void) fscanf(f, "%*f %*f %*f %*f %*f %*f %*f %*f %*f %*f\n");
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/ph/ssd-120/darktable/master/tools/noise/noiseprofile.c:90:12: warning: ignoring return value of ‘fscanf’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
   90 |     (void) fscanf(f, "%*f %*f %*f %*f %*f %*f %*f %f %f %f\n", hist + 3*k, hist+3*k+1, hist+3*k+2);
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Steps to reproduce

Compile current master locally

### Expected behavior

Don't throw warning when compiling 

### Logfile | Screenshot | Screencast

_No response_

### Commit

_No response_

### Where did you obtain darktable from?

self compiled

### darktable version

5.5.0+739~g959c8b4160

### What OS are you using?

Linux

### What is the version of your OS?

Ubuntu 24.04

### Describe your system

OS : Linux 6.8.0-106-generic / Ubuntu 24.04.4 LTS                                                              
Memory : MemTotal:       65759236 kB                                                                           
Graphics Card :     Product Name                                       : NVIDIA GeForce RTX 2060               
Graphics Card : Driver Version                                         : 580.126.09                            
OpenCL installed :   Device OpenCL C Version                         OpenCL C 1.2                              
OpenCL activated : yes                                                                                         
Xorg : Version: 1:7.7+23ubuntu3                                                                                
Desktop : LXQt                                                                                                 
GTK+ : 3.24.41                                                                                                 
Glib : ldd (Ubuntu GLIBC 2.39-0ubuntu8.7) 2.39                                                                 
gcc (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0                                                                    
CMAKE_BUILD_TYPE : RelWithDebInfo

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

--- a/tools/noise/noiseprofile.c
+++ b/tools/noise/noiseprofile.c
@@ -59,11 +59,11 @@ read_pfm(const char *filename, int *wd, int*ht)
 {
   FILE *f = fopen(filename, "rb");
   if(!f) return 0;
-  if(fscanf(f, "PF\n%d %d\n%*[^\n]", wd, ht) < 2) return;
+  if(fscanf(f, "PF\n%d %d\n%*[^\n]", wd, ht) < 2) return NULL;
   fgetc(f); // eat only one newline
 
   float *p = (float *)malloc(sizeof(float)*3*(*wd)*(*ht));
-  if(fread(p, sizeof(float)*3, (*wd)*(*ht), f) == 0) return;
+  if(fread(p, sizeof(float)*3, (*wd)*(*ht), f) == 0) { free(p); return NULL; }
   for(int k=0;k<3*(*wd)*(*ht);k++) p[k] = fmaxf(0.0f, p[k]);
   fclose(f);
   return p;
@@ -149,6 +149,7 @@ int main(int argc, char *arg[])
   }
   int wd, ht;
   float *input = read_pfm(arg[1], &wd, &ht);
+  if(!input) { fprintf(stderr, "error: could not read input file '%s'\n", arg[1]); return 1;}
   float max = 0.0f;
   // sanity checks:
   // for(int k=0;k<3*wd*ht;k++) input[k] = clamp(input[k], 0.0f, 1.0f);

